File: src/main/java/org/telran/lecture_03_recursion/home_work/hanoi.js
function hanoi(n, start, end, aux) {
  // n: количество дисков
  // start: начальный столбец
  // end: конечный столбец
  // aux: вспомогательный столбец

  if (n === 1) {
    console.log(`${start} --> ${end}`);
    return;
  }
  hanoi(n - 1, start, aux, end);
  console.log(`${start} --> ${end}`);
  hanoi(n - 1, aux, end, start);
}

hanoi(3, "A", "C", "B"); // Выводит в терминал описание действие по перемещению дисков "A" --> "C"

File: src/main/java/org/telran/lecture_04_d_and_c/examples/Hanoi.java
package org.telran.lecture_04_d_and_c.examples;

public class Hanoi {

    public static void hanoi(int n, char source, char destination, char helper) {
        // n: количество дисков
        // start: начальный столбец
        // end: конечный столбец
        // aux: вспомогательный столбец
        if (n > 0) {
            // Перемещаем n-1 дисков с исходного на вспомогательный
            hanoi(n - 1, source, helper, destination);

            // Перемещаем самый большой диск с исходного на целевой
            System.out.println("Переместить диск " + n + " с " + source + " на " + destination);

            // Перемещаем n-1 дисков с вспомогательного на целевой
            hanoi(n - 1, helper, destination, source);
        }
    }

    public static void main(String[] args) {
        hanoi(4, 'A', 'C', 'B');
    }
}